agent/grpc/public/services/peerstream/replication.go

```diff
@@ -240,12 +241,48 @@ func (s *Server) handleUpdateService(
 	sn structs.ServiceName,
 	pbNodes *pbservice.IndexedCheckServiceNodes,
 ) error {
+	_, peering, err := s.GetStore().PeeringRead(nil, state.Query{Value: peerName})
+	if err != nil {
+		return fmt.Errorf("failed to read peering: %w", err)
+	}
+	// an invariant here should be that a peering exists for PeeringRead
+	if peering == nil {
+		return fmt.Errorf("did not find peering %q", peerName)
+	}
+
+	logger := s.Logger.With("peerName", peerName, "peerID", peering.ID)
+
 	// Capture instances in the state store for reconciliation later.
 	_, storedInstances, err := s.GetStore().CheckServiceNodes(nil, sn.Name, &sn.EnterpriseMeta, peerName)
 	if err != nil {
 		return fmt.Errorf("failed to read imported services: %w", err)
 	}
 
+	// account for imported services
+	if len(storedInstances) > 0 && len(pbNodes.GetNodes()) < 1 {
```

Comment:
`len(pbNodes.GetNodes()) == 0` could be a bit clearer here

The [Go language spec](https://go.dev/ref/spec#Array_types) dictates the length of an array type won't be negative